src/main/java/org/threadly/concurrent/future/FutureUtils.java

```diff
@@ -1441,6 +1441,160 @@ public void handleResult(ST result) {
       return slf;
     }
   }
+
+  /**
+   * Transform a future's failure condition into another future by applying the transformation 
+   * function.  The mapper can then choose to either throw an exception, or convert the exception 
+   * back into a normal result.
+   * TODO verify javadocs below  
+   * <p>
+   * This can be easily used to chain together a series of operations, happening async until the 
+   * final result is actually needed.  Once the future completes the mapper function will be invoked 
+   * on the executor (if provided).  Because of that providing an executor can ensure this will 
+   * never block.  If an executor is not provided then the mapper may be invoked on the calling 
+   * thread (if the future is already complete), or on the same thread which the future completes 
+   * on.  If the mapper function is very fast and cheap to run then {@link #map(Function)} or 
+   * providing {@code null} for the executor can allow more efficient operation.
+   * 
+   * @since 5.17
+   * @param <ST> The source type for the object returned from the future and inputed into the mapper
```

Comment:
`TT`?

Reply:
I had not completed the javadocs, thanks for pointing this out (I forgot)